### DISCUSSION TITLE
valueTransformer empty to null value

### DISCUSSION THREAD

**Discussion by marco-origam** [danvick/flutter_form_builder#1286] [Q&A]
I need the field to be null when it's empty like below:

```javascript
{
  name: null
}
``` 

So, im using valueTransformer as follows:

```dart
...
name: 'name',
valueTransformer: (value) {
  if (value == null || value.trim().isEmpty) {
    return null;
  }
  return value;
},
...
``` 

But, im still getting:

```javascript
{
  name: ""
}
``` 

Is there a way i can achieve it?

**Comment by Schrolli91** [comment]
I have the same Problem here...
```dart
valueTransformer: (String? value) {
  if (value == null) return null;
  return double.tryParse(value);
},
```
but the field is always exact the data, were i have written into the field.
I read with
```dart
if (formKey.currentState!.saveAndValidate()) {
  dynamic data = formKey.currentState?.value;
```

**Reply by Schrolli91** [reply]
Or is it possible to get the transformed values this way?
`formKey.currentState!.getTransformedValue('quantity')`
Can't find documentation about that.

**Reply by Schrolli91** [reply]
Ok i see - the Problem is the valueTransformer implementation.
If i return a `null` the original value will be used. A Bug in my opinion, because null can be a valid value.
`dynamic get transformedValue => widget.valueTransformer?.call(value) ?? value;`
